### PR TITLE
chore: automate adding sponsors section to readme

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What changes did you make? (Give an overview)
- added update-readme file
- added commit-readme file

#### Related Issues

fixes https://github.com/eslint/rewrite/issues/54

#### Is there anything you'd like reviewers to focus on?

- made the file ESM as we use type module in package.json
- should we add technology sponsors as well? 
- should we add team section as well? 
- allow the cron to update the readme right from first time
- secret token should be available by default? (`WORKFLOW_PUSH_BOT_TOKEN`)

### CODE REVIEW

File: tools/update-readme.js
Comment (mdjermanovic) [inline tools/update-readme.js:15]:
```suggestion
import { readFileSync, writeFileSync } from "node:fs";
```

To fix the lint error.

File: README.md
Comment (nzakas) [inline README.md:15]:
```suggestion
The following companies, organizations, and individuals support ESLint's ongoing maintenance and development. [Become a Sponsor](https://eslint.org/donate) to get your logo on our README and website.
```

File: packages/compat/README.md
Comment (nzakas) [inline packages/compat/README.md:192]:
```suggestion
The following companies, organizations, and individuals support ESLint's ongoing maintenance and development. [Become a Sponsor](https://eslint.org/donate) to get your logo on our README and website.
```

File: packages/config-array/README.md
Comment (nzakas) [inline packages/config-array/README.md:344]:
```suggestion
The following companies, organizations, and individuals support ESLint's ongoing maintenance and development. [Become a Sponsor](https://eslint.org/donate) to get your logo on our README and website.
```

File: packages/migrate-config/README.md
Comment (nzakas) [inline packages/migrate-config/README.md:79]:
```suggestion
The following companies, organizations, and individuals support ESLint's ongoing maintenance and development. [Become a Sponsor](https://eslint.org/donate) to get your logo on our README and website.
```

File: packages/object-schema/README.md
Comment (nzakas) [inline packages/object-schema/README.md:228]:
```suggestion
The following companies, organizations, and individuals support ESLint's ongoing maintenance and development. [Become a Sponsor](https://eslint.org/donate) to get your logo on our README and website.
```

File: tools/commit-readme.sh
Comment (nzakas) [inline tools/commit-readme.sh:14]:
```suggestion
	git commit -m "docs: Update README sponsors"
```

File: tools/update-readme.js
Comment (nzakas) [inline tools/update-readme.js:26]: I think we can remove all of the team-related functionality from this file. Let's just focus on sponsors.

File: tools/update-readme.js
Comment (nzakas) [inline tools/update-readme.js:4]: Not true.

File: tools/update-readme.js
Comment (nzakas) [inline tools/update-readme.js:28]: Can we autogenerate this so we don't need to keep updating?

I think we can hardcode `"./README.md"` and then just add every directory under `./packages`.

**Review by nzakas — CHANGES_REQUESTED** [review]
Thanks for getting this started. I left some suggestions to simplify things.

File: tools/update-readme.js
Comment (harish-sethuraman) [reply tools/update-readme.js:24]: Done

File: tools/update-readme.js
Comment (harish-sethuraman) [reply tools/update-readme.js]: changed

File: tools/commit-readme.sh
Comment (harish-sethuraman) [reply tools/commit-readme.sh]: updated

File: README.md
Comment (harish-sethuraman) [reply README.md]: changed all occurences. Shall I send a PR to update the same in eslint/eslint repo as well?

File: tools/commit-readme.sh
Comment (harish-sethuraman) [inline tools/commit-readme.sh:13]: changed from `git add **/**/README.md` as it would reference `node_modules` as well

File: package.json
Comment (mdjermanovic) [inline package.json:30]: This doesn't seem necessary?

Comment (harish-sethuraman) [reply package.json:30]: I can see that at lot of places there are mentions of the same package like. But I wonder how this appeared out of no where. I never installed it 🤔 

https://github.com/eslint/rewrite/blob/19a755b85e4d1f9299b32293e12221406b0c1abc/packages/migrate-config/tests/fixtures/reveal-md/expected.mjs#L7

Comment (harish-sethuraman) [reply package.json:30]: It was removed in https://github.com/eslint/rewrite/pull/43/files 🤔 how is it getting referenced? I don't see it as internal package as well?

Comment (harish-sethuraman) [reply package.json:30]:
```suggestion
```
feel free to commit this and merge in case its not required

File: .github/workflows/update-readme.yml
Comment (nzakas) [inline .github/workflows/update-readme.yml:5]: If we add `workflow_dispatch` then we can [run this manually](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/):

```suggestion
on:
    schedule:
        - cron: "0 8 * * *" # Every day at 1am PDT
    workflow_dispatch:
```

Comment (mdjermanovic) [reply .github/workflows/update-readme.yml:5]: Sounds good to me. @harish-sethuraman what do you think about committing this change?

Comment (harish-sethuraman) [reply .github/workflows/update-readme.yml:5]: Done. Seems like this should work. I don't have write access to this repo so I would not be able to trigger workflow manually?

Comment (mdjermanovic) [reply .github/workflows/update-readme.yml:5]: I think it can be triggered on the default (`main`) branch only. We can try it out when this PR is merged.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @nzakas to verify.

File: .github/workflows/update-readme.yml
Comment (nzakas) [reply .github/workflows/update-readme.yml:5]: Yes that's correct, it can only be triggered on the `main` branch once its merged.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**
